dask_ml/model_selection/_incremental.py

```diff
@@ -958,6 +951,25 @@ def __init__(
         )
 
     def fit(self, X, y=None, **fit_params):
+        if "IncrementalSearchCV" in str(type(self)):
```

Comment:
Mmm let's avoid the `str(type(self))` stuff. A little less direct, but how about you define a `_check_decay_deprecation()` method

```python
def _check_decay_deprecation(self):
     if self.decay_rate is None: ...
```

then in the subclass (`InverseDecaySearchCV`?) define that method to do nothing.